### PR TITLE
#54: Reveal describe details (tag name, number of commits)

### PR DESCRIPTION
**Pull request by TheSnoozer**
Hi @ktoso,
today I had some time to implement the feature for #54.
This PR looks huge but it isn't.
Most what I did was shifting code from DescribeCommand.java to a new class JGitCommon.java to make it easier accessible. This basically gave me the opportunity to reuse the major part of the functionality. We still could speed up the JGit-Implementation by saving the values we need twice but I leave this as optional :-9

Also I fixed the license-headers (some files had a duplicated license info) and added the new properties to the readme (I caught one that was missing...).

Let me know if I should improve something for the PR.

Thanks,

### CODE REVIEW

**Comment by ktoso** [comment]
Sounds great, reviewing!

File: src/main/java/pl/project13/jgit/JGitCommon.java [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+import com.google.common.collect.Lists;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static com.google.common.collect.Lists.newLinkedList;
+import static com.google.common.collect.Maps.newHashMap;
+import static com.google.common.collect.Sets.newHashSet;
+
+
+/**
+ * @author <a href="mailto:konrad.malawski@java.pl">Konrad 'ktoso' Malawski</a>
```
Comment (ktoso) [inline src/main/java/pl/project13/jgit/JGitCommon.java]: Let's remove all `@author` tags by the way, to make nier to feel collective ownership about the code. I'll do that. :-)

File: src/main/java/pl/project13/jgit/dummy/DatedRevTag.java [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
+ * git-commit-id-plugin is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with git-commit-id-plugin.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
 package pl.project13.jgit.dummy;
```
Comment (ktoso) [inline src/main/java/pl/project13/jgit/dummy/DatedRevTag.java]: thanks

File: src/main/java/pl/project13/maven/git/JGitProvider.java [hunk 3]
```diff
@@ -175,46 +193,40 @@ protected String getRemoteOriginUrl() throws MojoExecutionException {
...
-      });
-
-      Collection<String> tags = Collections2.transform(tagsForHeadCommit, new Function<Ref, String>() {
-        @Override public String apply(Ref input) {
-          return input.getName().replaceAll("refs/tags/", "");
-        }
-      });
-
+      ObjectId headId = headCommit.toObjectId();
+      Collection<String> tags = new JGitCommon().getTags(repo,headId);
```
Comment (ktoso) [inline src/main/java/pl/project13/maven/git/JGitProvider.java]: Why not reuse one instance? Is it stateful so it can't just be static methods?

Comment (ktoso) [inline src/main/java/pl/project13/maven/git/JGitProvider.java]: I'd opt for making it a constructor argument for each class needing it I guess.

**Comment by TheSnoozer** [comment]
Well I think we could make it static.
In one of my projects I learned the hard way to avoid static method calls and rather go for new Clazz().method ;-)
I think this memory is burned into my head :-)

**Comment by ktoso** [comment]
Or better just pass the instance around ;)

**Comment by TheSnoozer** [comment]
Mhh I  think I would leave it as it is right now :-)
Passing only a single instance around or making the calls static is something I would leave optional for further improvement :-P

**Comment by ktoso** [comment]
Though I would much rather have it passed in from the "beginning of the control flow" (the plugin, or the test) let's go with the current impl for now ;-)

**Comment by ktoso** [comment]
Thanks, merged!

**Pull request merged**